tests/Integration/LookupTest.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Automattic\LegacyRedirector\Tests\Integration;
+
+use Automattic\LegacyRedirector\Lookup;
+use WPCOM_Legacy_Redirector;
+
+/**
+ * CapabilityTest class.
+ */
+final class LookupTest extends TestCase {
+	/**
+	 * Test Lookup::get_redirect_uri.
+	 *
+	 * @covers Lookup::get_redirect_uri
+	 * @dataProvider get_protected_redirect_data
+	 *
+	 * @param string $from_url        Redirect From URL.
+	 * @param string $to_url          Redirect To URL.
+	 * @param int    $redirect_status Redirect Status Code.
+	 * @return void
+	 */
+	public function test_get_redirect_uri( $from_url, $to_url, $redirect_status ) {
+
+		$this->assertTrue( WPCOM_Legacy_Redirector::insert_legacy_redirect( $from_url, $to_url, false ) );
+
+		$redirect_data = Lookup::get_redirect_data( $from_url );
+
+		$this->assertEquals( $to_url, $redirect_data['redirect_uri'] );
+		$this->assertEquals( $redirect_status, $redirect_data['redirect_status'] );
+
+	}
+	/**
+	 * Data provider for tests methods
+	 *
+	 * @return array
+	 */
+	public function get_protected_redirect_data() {
+		return array(
+			'redirect_unicode_with_query' => array(
```

Comment:
As per Example 2.6 at https://phpunit.readthedocs.io/en/9.5/writing-tests-for-phpunit.html#data-providers named providers don't need to be single underscored words.

```suggestion
			'redirect unicode characters with querystring' => array(
```

The other dataProviders in this PR should be adjusted to lose the `_` as well.